cli/internal/daemon/daemon.go

```diff
@@ -0,0 +1,324 @@
+package daemon
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"sync"
+	"time"
+
+	grpc_recovery "github.com/grpc-ecosystem/go-grpc-middleware/recovery"
+	"github.com/hashicorp/go-hclog"
+	"github.com/mitchellh/cli"
+	"github.com/nightlyone/lockfile"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/vercel/turborepo/cli/internal/config"
+	"github.com/vercel/turborepo/cli/internal/daemon/connector"
+	"github.com/vercel/turborepo/cli/internal/fs"
+	"github.com/vercel/turborepo/cli/internal/server"
+	"github.com/vercel/turborepo/cli/internal/signals"
+	"github.com/vercel/turborepo/cli/internal/util"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// Command is the wrapper around the daemon command until we port fully to cobra
+type Command struct {
+	Config        *config.Config
+	UI            cli.Ui
+	SignalWatcher *signals.Watcher
+}
+
+// Run runs the daemon command
+func (c *Command) Run(args []string) int {
+	cmd := getCmd(c.Config, c.UI, c.SignalWatcher)
+	cmd.SetArgs(args)
+	err := cmd.Execute()
+	if err != nil {
+		return 1
+	}
+	return 0
+}
+
+// Help returns information about the `daemon` command
+func (c *Command) Help() string {
+	cmd := getCmd(c.Config, c.UI, c.SignalWatcher)
+	return util.HelpForCobraCmd(cmd)
+}
+
+// Synopsis of daemon command
+func (c *Command) Synopsis() string {
+	cmd := getCmd(c.Config, c.UI, c.SignalWatcher)
+	return cmd.Short
+}
+
+type daemon struct {
+	logger     hclog.Logger
+	repoRoot   fs.AbsolutePath
+	timeout    time.Duration
+	reqCh      chan struct{}
+	timedOutCh chan struct{}
+	cleanup    sync.Once
+}
+
+func getRepoHash(repoRoot fs.AbsolutePath) string {
+	pathHash := sha256.Sum256([]byte(repoRoot.ToString()))
+	// We grab a substring of the hash because there is a 108-character limit on the length
+	// of a filepath for unix domain socket.
+	return hex.EncodeToString(pathHash[:])[:16]
+}
+
+func getDaemonFileRoot(repoRoot fs.AbsolutePath) fs.AbsolutePath {
+	tempDir := fs.TempDir("turbod")
```

Comment:
Design opinion: I feel like this should go in `/run` or somewhere else that seems a bit more connected to our stuff, not in `/tmp`. That it doesn't persist through reboots is a clever hack but I don't think it's worth it.